### ISSUE TITLE
[Pricing Table & Structured List]: QA: JAWS with FF does not announce the first Row header

### ISSUE THREAD

**Issue by gopinathan-n** [carbon-design-system/carbon-for-ibm-dotcom#9173]
### Description

Page Element: Row header announcement

Issue Description:
VO iPhone with Safari announces the first Row header. However JAWS with FF does not announce the first Row header

Expected:
Screen reader is expected to identify the row header correctly.

User Impact:
If the table structure is not conveyed properly, it’s difficult for screen reader users to navigate through the table content and correlate data. 

### Component(s) impacted

Pricing table default
Structured List default

### Browser

Firefox

### Carbon for IBM.com version

v1.36.0-rc.0

### Severity

Severity 3 = The problem is visible or noticeable to users but does not impede the usability or functionality. Affects minor functionality, has a workaround.

### Application/website

Carbon for IBM.com

### Package

@carbon/ibmdotcom-web-components

### CodeSandbox example

https://carbon-design-system.github.io/carbon-for-ibm-dotcom/canary/web-components-experimental

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [ ] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues
